packages/http-client-csharp/generator/Microsoft.TypeSpec.Generator.ClientModel/src/Providers/ScmMethodProviderCollection.cs

```diff
@@ -702,7 +702,7 @@ private CSharpType GetConvenienceReturnType(IReadOnlyList<InputOperationResponse
                 var type = (response?.BodyType as InputModelType)?.Properties.FirstOrDefault(p =>
                     p.SerializedName == _pagingServiceMethod.PagingMetadata.ItemPropertySegments[0]);
 
-                responseBodyType = response?.BodyType is null || type is null ? null : GetResponseBodyType((type.Type as InputArrayType)!.ValueType);
+                responseBodyType = response?.BodyType is null || type?.Type is not InputArrayType arrayType ? null : GetResponseBodyType(arrayType.ValueType);
```

Comment:
This is a temporary workaround until we have support for nested response paging.